gen3/tools/indexing/post_indexing_validation.py

```
@@ -0,0 +1,250 @@
+
+        Args:
+            gen3file (object): An object that provides the `get_presigned_url` method to generate a pre-signed URL for the record.
+
+        Returns:
+            None: This function does not return any value. It modifies the `response_status` and `download_status` attributes.
+        """
+        logger.info(f"Checking record {self.guid}")
+        try:
+            resp = gen3file.get_presigned_url(self.guid)
+            url = resp.get("url")
+            response_status = 200
+            logger.info(
+                f"Pre-signed url successfully generated for record {self.guid} with status code {response_status}"
+            )
+        except requests.HTTPError as err:
+            response_status = err.response.status_code
+            logger.info(f"Pre-signed url generation failed for record {self.guid}")
+        self.response_status = response_status
+
+        download_success = -1
+        if response_status == 200:
+            try:
+                download_success = requests.get(url).status_code
+                logger.info(
+                    f"Download process complete with status code {download_success}"
+                )
+            except:
+                download_success = -1
+        self.download_status = download_success
+        return
+
+
+class Records:
+    def __init__(self, auth):
+        self.auth = auth
+        self.access_token = auth.get_access_token()
+        self.commons = auth.endpoint
+        self.record_dict = {}
+        self.record_sizes = {}
+        self.headers = {
+            "accept": "application/json",
+            "authorization": f"bearer {self.access_token}",
+        }
+
+    def read_records_from_manifest(self, manifest):
+        """
+        Parses a manifest and creates a dictionary of Record objects.
+
+        Args:
+            manifest (str): the location of a manifest file
+        """
+        if manifest[-3:] == "tsv":
```

Might want to validate the length of the manifest before doing this just in case it's empty or some such and then throw a more helpful error rather than the default invalid index you'd get.

---

Will update the validation more here good catch!